logbook-core/src/main/java/org/zalando/logbook/core/DefaultLogbook.java

```diff
@@ -44,7 +45,8 @@ public RequestWritingStage process(final HttpRequest originalRequest, final Stra
             final Precorrelation precorrelation = newPrecorrelation(originalRequest);
             final HttpRequest processedRequest = strategy.process(originalRequest);
 
-            final HttpRequest request = new CachingHttpRequest(processedRequest);
+            final HttpAttributes httpAttributes = strategy.getRequestAttributesExtractor().extractOrEmpty(processedRequest);
```

Comment:
This is the only place that actually needs the `orEmpty` semantic. I would move it here.

Reply:
Moved it here as per suggestion.